### PR TITLE
api: remove /v1/introspect for now

### LINKED ISSUE #1425: api: replace `GET /introspect` with `GET /identities/self`
The `/introspect` API endpoint is quite similar to `GET /identities/:id`. I think we could remove it by accepting a `self` keyword in place of the ID. That would allow a user to inspect their identity.

We may need to prevent anyone from creating an `Identity`  with an `ID` of `self` (if we don't already).

### PR DESCRIPTION
**Pull request by dnephin**
Branched from #1738, that will need to merge first
 
## Summary

This PR writes a bunch of tests for `GetIdentity` and adds support for `/v1/identities/self` to query for details about the current identity. By supporting `self` we remove the need for `/v1/introspect` for now, because all it returns is less information about the identity.

It is quite possible that in the future we will re-introduce an endpoint similar to `/v1/introspect`. By removing it now we won't have to worry about changing the structure of the response. We'll have more information about the use case it is intended to solve, which lets us design the endpoint from scratch.

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [ ] Updated associated configuration where necessary
- [ ] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #1425

### CODE REVIEW

File: internal/server/server_test.go [hunk 1]
```diff
@@ -197,6 +197,9 @@ func TestSignupEnabled(t *testing.T) {
 }
 
 func TestServer_Run(t *testing.T) {
+	if testing.Short() {
```
Comment (jmorganca) [inline internal/server/server_test.go:200]: How "slow" is this?

Comment (jmorganca) [inline internal/server/server_test.go:200]: How "slow" is this?

**Review by jmorganca — APPROVED** [review]

File: docs/api/openapi3.json [hunk 2]
```diff
@@ -2203,14 +2187,14 @@
...
         "parameters": [
           {
-            "example": "4yJ3n3D8E2",
             "in": "path",
             "name": "id",
             "required": true,
             "schema": {
+              "description": "a uid or the literal self",
               "example": "4yJ3n3D8E2",
               "format": "uid",
```
Comment (BruceMacD) [inline docs/api/openapi3.json:2196]: I think this format specifier should be `"uid|self" too

**Review by BruceMacD — APPROVED** [review]

File: docs/api/openapi3.json (hunk 2 above)
Comment (dnephin) [reply docs/api/openapi3.json:2196]: I'm not exactly sure how to represent a description of a format vs a literal, I'll use `uid|self` for now and we can figure it out later.

**Review by jmorganca — APPROVED** [review]

**Pull request merged**